kaldb/src/main/java/com/slack/kaldb/logstore/search/KaldbDistributedQueryService.java

```diff
@@ -361,8 +371,17 @@ public KaldbSearch.SearchResult doSearch(KaldbSearch.SearchRequest request) {
                   new SearchResultAggregatorImpl<>(SearchResultUtils.fromSearchRequest(request)))
               .aggregate(searchResults);
 
-      distributedQueryTotalNodes.increment(aggregatedResult.totalNodes);
-      distributedQueryFailedNodes.increment(aggregatedResult.failedNodes);
+      // We report a query with more than 0% of requested nodes, but less than 2% as a tolerable
+      // response. Anything over 2% is considered an unacceptable.
+      if (aggregatedResult.totalNodes == 0 || aggregatedResult.failedNodes == 0) {
```

Comment:
`aggregatedResult.totalNodes == 0` is a good check so that the remaining if-else doesn't get into funky divisions

Reply:
Indeed. 

Theoretically we should never have non-zero failed nodes and zero total nodes, but it doesn't hurt to check. 😄